shell/platform/embedder/embedder.cc

```diff
@@ -993,13 +994,42 @@ FlutterEngineResult FlutterEngineInitialize(size_t version,
                               "Compositor arguments were invalid.");
   }
 
+  auto display_settings_callback =
+      SAFE_ACCESS(args, display_settings_callback, nullptr);
+  double main_display_refresh_rate =
+      flutter::VsyncWaiter::kUnknownRefreshRateFPS;
+
+  if (display_settings_callback) {
+    FlutterDisplaySettings display_settings;
+    display_settings.struct_size = sizeof(FlutterDisplaySettings);
+    display_settings.displays = nullptr;
+    display_settings.display_count = 0;
+    display_settings_callback(user_data, &display_settings);
+
+    if (display_settings.display_count == 0) {
+      return LOG_EMBEDDER_ERROR(kInvalidArguments,
+                                "Invalid display settings provided by the "
+                                "embedder, got 0 active displays.");
+    }
+
+    display_settings.displays =
+        new FlutterDisplay[display_settings.display_count];
```

Comment:
Could we allocate this as a `std::vector` then hand the callback `displays.data()` instead to avoid dealing with new/delete?

Reply:
Good idea!